styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -330,7 +329,9 @@ public void create(Environment environment) {
     final Config staleStateTtlConfig = config.getConfig(STYX_STALE_STATE_TTL_CONFIG);
     final TimeoutConfig timeoutConfig = TimeoutConfig.createFromConfig(staleStateTtlConfig);
 
-    warmUpCache(workflowCache, storage);
+    final WorkflowCache workflowCache = new InMemWorkflowCache(
+        new CachedSupplier<>(storage::workflows, time));
+
     initializeResources(storage, shardedCounter, counterSnapshotFactory, timeoutConfig, workflowCache);
```

Comment:
If this workflow cache is only used during initialization, how about just reading the all the workflows from storage here and directly passing that in to `initializeResources` etc instead of having the `WorkflowCache` layer of indirection?

Reply:
It is used as a `Gauge` for metrics and that is why we chose `CachedSupplier`.

Reply:
Ah, the cache os also passed into `setupMetrics` and used there in gauge lambas so just reading the workflows once doesn't work.

I guess the `WorkflowCache` could still be replaced with a `Supplier<Map<WorkflowId, Workflow>>`, but maybe it's convenient to keep the `WorkflowCache` interface?

Reply:
Yeah, right. Mostly our laziness. Didn't want to change too many things where that interface is expected. :D

Reply:
👍